added BBDC support statements/language (#884)
hashicorp/terraform-website: content/source/docs/cloud/vcs/bitbucket-server.html.md

---
layout: "cloud"
page_title: "Bitbucket Server and Data Center - VCS Providers - Terraform Cloud"
---

# Configuring Bitbucket Server/Data Center Access

> These instructions are for using Bitbucket Server for Terraform Cloud's VCS features. Bitbucket Server is the on-premises version of Bitbucket; [Bitbucket Cloud has separate instructions,](./bitbucket-cloud.html) as do the [other supported VCS providers.](./index.html)

Reviewer: ```suggestion
These instructions are for using **Bitbucket Server** for Terraform Cloud's VCS features. 

These instructions also apply to **Bitbucket Data Center,** which is a variant of Bitbucket Server that supports clustering. Terraform Cloud treats these two products identically, and Bitbucket Data Center users will select **Bitbucket Server** as their VCS Provider type. Unless stated otherwise, any reference to Bitbucket Server in this document also applies to Bitbucket Data Center.

[Bitbucket Cloud has separate instructions,](./bitbucket-cloud.html) as do the [other supported VCS providers.](./index.html)
```

Author: You say 'unless stated otherwise' but I don't see anywhere that we actually state otherwise?

Reviewer: The only spot is in the version note, which has a stricter requirement if you want to use BBDC.

Note that Bitbucket Server requires both OAuth authentication and an SSH key. The instructions below include SSH key configuration.

> -> **Version note:** Terraform Cloud supports Bitbucket Server versions 4.9.1 and newer. HashiCorp does not test older versions of Bitbucket Server with Terraform Cloud, and they might not work as expected. Also note that, although we do not deliberately remove support for versions that have reached end of life (per the [Atlassian Support End of Life Policy](https://confluence.atlassian.com/support/atlassian-support-end-of-life-policy-201851003.html)), our ability to resolve customer issues with end of life versions might be limited.

Reviewer: ```suggestion
-> **Version note:** Terraform Cloud supports Bitbucket Server versions 4.9.1 and newer, and Bitbucket Data Center versions 5.4.0 and newer. HashiCorp does not test older versions of Bitbucket Server with Terraform Cloud, and they might not work as expected. Also note that, although we do not deliberately remove support for versions that have reached end of life (per the [Atlassian Support End of Life Policy](https://confluence.atlassian.com/support/atlassian-support-end-of-life-policy-201851003.html)), our ability to resolve customer issues with end of life versions might be limited.
```

Author: I think this is actually confusing as it doesn't draw a distinction between Bitbucket Server (legacy) and Bitbucket Server (native - new) - we have two distinct repo types now, because of technical differences between them - and we have explicit version statements for both, one of which allows us to use Data Center and one of which does not. I think the prior support statement I wrote was clearer as it laid.

Reviewer: I'm not sure what you mean here; the rest of the docs don't draw a named distinction between the two, and the UI when creating the connection only asks whether your version number is over the mark. 

The reason I found your original wording confusing was this part: 

> Bitbucket Data Center users will follow the same instructions as a user with Bitbucket Server **with version greater than v5.4.0**, omitting the installation of the Web Post Hooks add-on.

That reads as "any BBDC user should behave as though they were on BBS version 5.4+," when in fact it's sort of the opposite: BBDC users must be on version 5.4+, and then should just follow the appropriate instructions.

Author: OK, I can see how you would think that. This seems OK to me if you are satisfied with it. Given this much more streamlined wording, do you still want to change the title of the page and sidebar nav link or no?

Reviewer: Yeah, go ahead and leave them as "server and data center." That way users won't have any doubt about which page they need to go to, and once they're there they should be able to understand the situation.

> -> **Bitbucket Data Center Support:** Please note that these instructions are equally applicable to both Bitbucket Server and Bitbucket Data Center. Bitbucket Data Center users will follow the same instructions as a user with Bitbucket Server **with version greater than v5.4.0**, omitting the installation of the Web Post Hooks add-on. You will still select **Bitbucket Server** as the VCS Provider type.

Reviewer: ```suggestion
```

Author: See my following comment; I think we should maintain this as I wrote it

> 

Reviewer: ```suggestion
```

---
layout: "cloud"
page_title: "Bitbucket Server and Data Center - VCS Providers - Terraform Cloud"
---

# Configuring Bitbucket Server/Data Center Access

These instructions are for using Bitbucket Server/Data Center for Terraform Cloud's VCS features. Bitbucket Server/Data Center are the on-premises versions of Bitbucket; [Bitbucket Cloud has separate instructions,](./bitbucket-cloud.html) as do the [other supported VCS providers.](./index.html)

Note that Bitbucket Server/Data Center requires both OAuth authentication and an SSH key. The instructions below include SSH key configuration.

-> **Version note:** Terraform Cloud supports Bitbucket Server versions 4.9.1 and newer. HashiCorp does not test older versions of Bitbucket Server with Terraform Cloud, and they might not work as expected. Also note that, although we do not deliberately remove support for versions that have reached end of life (per the [Atlassian Support End of Life Policy](https://confluence.atlassian.com/support/atlassian-support-end-of-life-policy-201851003.html)), our ability to resolve customer issues with end of life versions might be limited.

> -> **Bitbucket Data Center Support:** Please note that these instructions are equally applicable to both Bitbucket Server and Bitbucket Data Center. Bitbucket Data Center users will follow the same instructions as a user with Bitbucket Server **with version greater than v5.4.0**, omitting the installation of the Web Post Hooks add-on. You will still select **Bitbucket Server** as the VCS Provider type.

Reviewer 2: Is this call out not sufficient? I don't love the `Bitbucket Server/Data Center` scattered throughout and wonder if this note on its own is enough to satisfy the needs of the addition. Presumably those that run BBDC as also aware that it is BitBucket Server and would find their way to this connection type without the modification to title and naming?

Author: I considered that, and am not particularly precious either way - I just wanted to make sure it was explicitly called out in at least one place for SEO/other discoverability purposes. Will gladly revert the other instances and leave it just in the title with this callout if others agree.

Reviewer 2: What does navigation look like with that long of a title? 
![image](https://user-images.githubusercontent.com/586829/63378270-81471080-c357-11e9-8ae7-0ccc68b3b68c.png)

Reviewer 2: ah, I guess the navigation link isn't being changed at the moment, so we'd look like this

![image](https://user-images.githubusercontent.com/586829/63378333-a6d41a00-c357-11e9-982c-6ec94f8b0741.png)

Author: I haven't seen it rendered but the sidebar has a good amount of padding around it, I don't expect it would be any wider than "Users, Teams, Organizations" below

Author: OK @caseylang take another look, I've reduced the changes now.

Reviewer 2: ![image](https://user-images.githubusercontent.com/586829/63380106-6aa2b880-c35b-11e9-9f01-cab4a1050327.png)

Keep in mind that your change currently does not modify the navigation link. To do that you would need to update the following as well: 
https://github.com/hashicorp/terraform-website/blob/b7b7e6c33c2f9092bb7620227207d47bf6785988/content/source/layouts/cloud.erb#L60-L62

Author: Good call, thanks - added to PR
